kubernetes/schema_job_spec.go

```diff
@@ -91,6 +91,70 @@ func jobSpecFields(specUpdatable bool) map[string]*schema.Schema {
 			ValidateFunc: validateNonNegativeInteger,
 			Description:  "Specifies the maximum desired number of pods the job should run at any given time. The actual number of pods running in steady state will be less than this number when ((.spec.completions - .status.successful) < .spec.parallelism), i.e. when the work left to do is less than max parallelism. More info: https://kubernetes.io/docs/concepts/workloads/controllers/jobs-run-to-completion/",
 		},
+		"pod_failure_policy": {
+			Type:        schema.TypeList,
+			Optional:    true,
+			ForceNew:    false,
+			MaxItems:    1,
+			Description: "Specifies the maximum desired number of pods the job should run at any given time. The actual number of pods running in steady state will be less than this number when ((.spec.completions - .status.successful) < .spec.parallelism), i.e. when the work left to do is less than max parallelism. More info: https://kubernetes.io/docs/concepts/workloads/controllers/jobs-run-to-completion/",
+			Elem: &schema.Resource{
+				Schema: map[string]*schema.Schema{
+					"rule": {
+						Type:        schema.TypeList,
+						Description: "A label query over volumes to consider for binding.",
+						Required:    true,
+						Elem: &schema.Resource{
+							Schema: map[string]*schema.Schema{
+								"action": {
+									Type:     schema.TypeString,
+									Optional: true,
+								},
+								"on_exit_codes": {
+									Type:     schema.TypeList,
+									Optional: true,
+									MaxItems: 1,
+									Elem: &schema.Resource{
+										Schema: map[string]*schema.Schema{
+											"container_name": {
+												Type:     schema.TypeString,
+												Optional: true,
+											},
+											"operator": {
+												Type:     schema.TypeString,
+												Optional: true,
+											},
+											"values": {
+												Type:     schema.TypeSet,
+												Optional: true,
+												MinItems: 1,
+												Elem:     &schema.Schema{Type: schema.TypeInt},
```

Comment:
We could probably do some more validation here. From the docs:

```
The list of values must be ordered and must not contain duplicates. Value '0' cannot be used for the In operator. At least one element is required. At most 255 elements are allowed.
```

So perhaps we should set a MaxItems of 255, and have a validate function that:
- Rejects a value of `0`
- Confirms the set of values is in order? Alternatively we could enforce the ordering by sorting it ourselves in the expander function.

Reply:
went ahead and added `slices.Sort()` in expander function